fondant/data_io.py

```diff
@@ -209,7 +207,11 @@ def create_subset_dataframe(
             subset_df = create_subset_dataframe(subset_name, subset_columns, df)
 
             # set index
-            subset_df.set_index("id")
+            if subset_df.index.name != "uid":
+                subset_df["uid"] = (
+                    subset_df["id"].astype("str") + "_" + subset_df["source"]
+                )
+                subset_df = subset_df.set_index("uid")
```

Comment:
Shouldn't we pass in the divisions so they are the same across index and all subsets?

Reply:
Any input on this @GeorgesLorre?

Reply:
I'm looking into this, can I take it up in a different PR ?

Reply:
Ok, I think we should still align them between the index and subsets as well, but let's do that as a follow up.